### PR TITLE
feat(perf-issues): Return perf problems from detection

### PR DESCRIPTION
**Pull request by udameli**

### CODE REVIEW

File: src/sentry/utils/performance_issues/performance_detection.py [hunk 1]
```diff
@@ -120,12 +146,62 @@ def _detect_performance_issue(data: Event, sdk_span: Any):
...
         for _, detector in detectors.items():
             detector.visit_span(span)
 
     report_metrics_for_detectors(event_id, detectors, sdk_span)
 
+    detected_problems = [
+        (i, detector_type)
+        for detector_type, d in used_perf_issue_detectors.items()
+        for _, i in d.stored_problems
+    ]
```
Comment (k-fish) [inline src/sentry/utils/performance_issues/performance_detection.py:164]: 🤔

File: src/sentry/utils/performance_issues/performance_detection.py [hunk 2]
```diff
@@ -260,14 +336,14 @@ class DuplicateSpanHashDetector(PerformanceDetector):
     Uses the span grouping strategy hash to potentially detect duplicate spans more accurately.
     """
 
-    __slots__ = ("cumulative_durations", "duplicate_spans_involved", "stored_issues")
+    __slots__ = ("cumulative_durations", "duplicate_spans_involved", "stored_problems")
```
Comment (k-fish) [inline src/sentry/utils/performance_issues/performance_detection.py:339]: 👍

**Review by k-fish — APPROVED** [review]
Yeah, it's prob better to base against master if those other changes weren't really necessary.

**Review by untitaker — APPROVED** [review]
i only reviewed the changes in event manager, no operational concerns... i suppose we have to write something back into the event at some point and that's why it's all in save_event rather than a separate task?

**Comment by udameli** [comment]
> i only reviewed the changes in event manager, no operational concerns... i suppose we have to write something back into the event at some point and that's why it's all in save_event rather than a separate task?

Thanks for the review! 

Yes, we'll need to write performance group ids to transactions since that's how we're linking affected transactions to performance issues

**Comment by udameli** [comment]
all changes were moved to https://github.com/getsentry/sentry/pull/38308. Closing this one

**Pull request closed without merging**